internal/clearingway/menu.go

```diff
@@ -0,0 +1,166 @@
+package clearingway
+
+import (
+	"fmt"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// guaranteed UI elements
+const(
+	MenuMain    string = "menucreate"
+	MenuVerify  string = "menuverify"
+	MenuRemove  string = "menuremove"
+)
+
+// menu types
+// currently only default/guaranteed menus, dropdown menu based on encounters
+type MenuType string
+const (
+	MenuTypeDefault   MenuType = "default"
+	MenuTypeEncounter MenuType = "encounter"
+)
+
+// struct to hold data for all different menu components
+type Menus struct {
+	Defaults map[string]*Menu
+	Menus    []*Menu
+}
+
+type Menu struct {
+	Name           string  // internal name e.g "menureclear"
```

Comment:
we might want to make the internal name more reader-friendly by either doing camel or snake case